### PR TITLE
refactor: clean up unwraps in wallet_ffi

### PR DESCRIPTION
**Pull request by StriderDM**
Description
---
Remove unwrap() in functions of wallet_ffi, code is considered best-effort to try to prevent a panic since the nature of handling invalid pointers or invalidly cast pointers passed in results in undefined behavior.

The function signature of log_debug_message() was modified to take into account an error occurring. PR is a breaking change due to this.

Fixed potential panic when unwrapping parse() for MultiAddr.

Fixed missing documentation for init_logging().

Motivation and Context
---
Removal of calls to unwrap() and bug fixes.

How Has This Been Tested?
---
cargo test --all

### CODE REVIEW

File: base_layer/wallet_ffi/src/error.rs [hunk 1]
```diff
@@ -101,6 +103,10 @@ impl From<InterfaceError> for LibWalletError {
                 code: 8,
                 message: "Balance Unavailable".to_string(),
             },
+            InterfaceError::PointerError(_) => Self {
+                code: 9,
+                message: format!("{:?}", v),
```
Comment (stringhandler) [inline base_layer/wallet_ffi/src/error.rs]:
```suggestion
                message: format!("Pointer error on {}:{:?}", p,  v),
```

File: base_layer/wallet_ffi/src/error.rs [hunk 2]
```diff
@@ -101,6 +103,10 @@ impl From<InterfaceError> for LibWalletError {
                 code: 8,
                 message: "Balance Unavailable".to_string(),
             },
+            InterfaceError::PointerError(_) => Self {
```
Comment (stringhandler) [inline base_layer/wallet_ffi/src/error.rs]:
```suggestion
            InterfaceError::PointerError(p) => Self {
```

**Review by hansieodendaal — APPROVED** [review]
Nice - LGTM (untested)

**Review by philipr-za — APPROVED** [review]
Looks good

**Pull request merged**